src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -332,6 +349,13 @@ else if (surrounded && isInTokenList(type, LITERALS)) {
                     log(ast, MSG_STRING,
                         chopString(ast.getText()));
                 }
+                else if (type == TokenTypes.TEXT_BLOCK_LITERAL_BEGIN) {
+                    // Strip newline control characters to keep message as single line.
+                    final String logString = NEWLINE.matcher(
+                        ast.getFirstChild().getText()).replaceAll("\\\\n");
+                    // Add quotes so that log message is consistent with STRING_LITERAL
+                    log(ast, MSG_STRING, QUOTE + logString + QUOTE);
```

Comment:
@pbludov @esilkensen I'm not sure whether we want to use the new utility introduced at https://github.com/checkstyle/checkstyle/pull/8705 to modify the string here, but I would still need to replace the newline control characters with the text "\n" to keep the error message to one line (like in our AST).  Let me know if we should do this or not, I can make the changes.

Reply:
In addition, the text should be truncated. Please note the `chopString` method in this check.